scripts/scil_compute_fodf.py

```diff
@@ -104,69 +99,25 @@ def main():
                                  args.frf_file])
     assert_outputs_exist(parser, args, arglist)
 
-    nbr_processes = args.nbr_processes
-    parallel = True
-    if nbr_processes is not None:
-        if nbr_processes <= 0:
-            nbr_processes = None
-        elif nbr_processes == 1:
-            parallel = False
-
     full_frf = np.loadtxt(args.frf_file)
-
-    if not full_frf.shape[0] == 4:
-        raise ValueError('FRF file did not contain 4 elements. '
-                         'Invalid or deprecated FRF format')
-
-    frf = full_frf[0:3]
-    mean_b0_val = full_frf[3]
-
     vol = nib.load(args.input)
-    data = vol.get_data()
-
+    data = vol.dataobj
     bvals, bvecs = read_bvals_bvecs(args.bvals, args.bvecs)
 
-    if not is_normalized_bvecs(bvecs):
-        logging.warning('Your b-vectors do not seem normalized...')
-        bvecs = normalize_bvecs(bvecs)
-
-    check_b0_threshold(args.force_b0_threshold, bvals.min())
-    gtab = gradient_table(bvals, bvecs, b0_threshold=bvals.min())
-
     if args.mask is None:
         mask = None
     else:
-        mask = nib.load(args.mask).get_data().astype(np.bool)
-
-    # Raise warning for sh order if there is not enough DWIs
-    if data.shape[-1] < (args.sh_order + 1) * (args.sh_order + 2) / 2:
-        warnings.warn(
-            'We recommend having at least {} unique DWIs volumes, but you '
-            'currently have {} volumes. Try lowering the parameter --sh_order '
-            'in case of non convergence.'.format(
-                (args.sh_order + 1) * (args.sh_order + 2) / 2, data.shape[-1]))
-
-    reg_sphere = get_sphere('symmetric362')
-    peaks_sphere = get_sphere('symmetric724')
-
-    csd_model = ConstrainedSphericalDeconvModel(
-        gtab, (frf, mean_b0_val),
-        reg_sphere=reg_sphere,
-        sh_order=args.sh_order)
-
-    peaks_csd = peaks_from_model(model=csd_model,
-                                 data=data,
-                                 sphere=peaks_sphere,
-                                 relative_peak_threshold=.5,
-                                 min_separation_angle=25,
-                                 mask=mask,
-                                 return_sh=True,
-                                 sh_basis_type=args.sh_basis,
-                                 sh_order=args.sh_order,
-                                 normalize_peaks=True,
-                                 parallel=parallel,
-                                 nbr_processes=nbr_processes)
+        mask = np.asanyarray(nib.load(args.mask_wm).dataobj).astype(np.bool)
```

Comment:
mask_wm does not exist

Reply:
I did not check all options/conditions, and I did not specifically provide a mask, which explains why my checks were passing. I am sorry @arnaudbore .